environments/servlet/tests/tomcat/pom.xml

```diff
@@ -158,6 +152,7 @@
                 <property>
                     <name>incontainer</name>
                 </property>
+                <jdk>(,1.8]</jdk>
```

Comment:
Not to activate this profile for JDK9+, i.e. activate only when incontainer property is set and JDK<=1.8. Without having this, both incontainer and jdk9 profiles are activated (mvn help:active-profiles). TBH I'm not sure what the result actually is when having both profiles activated because when running the tests, the result looks the same regardless of this line. I think it might be because the profiles are picked in some order - I just don't know if the order is well defined or if it is just a coincidence.

Reply:
You are right, the jdk condition is not needed here, thanks! The test exclusion is still needed though as related [Tomcat BZ](https://bz.apache.org/bugzilla/show_bug.cgi?id=57314) was closed as WONTFIX. The same test works on other servlet containers (e.g. Jetty) so we probably should not remove it entirely.